packages/compass-telemetry/src/telemetry-events.ts

```diff
@@ -0,0 +1,1435 @@
+type ConnectionScoped<E extends { payload: unknown }> = E & {
+  payload: E['payload'] & { connection_id: string };
+};
+
+/**
+ * This event is fired when user successfully signed in to their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignInSuccessEvent = {
+  name: 'Atlas Sign In Success';
+  payload: { auid: string };
+};
+
+/**
+ * This event is fired when user failed to sign in to their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignInErrorEvent = {
+  name: 'Atlas Sign In Error';
+  payload: { error: string };
+};
+
+/**
+ * This event is fired when user signed out from their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignOutEvent = {
+  name: 'Atlas Sign Out';
+  payload: { auid: string };
+};
+
+/**
+ * This event is fired when user selects a use case from the aggregation panel
+ *
+ * @category Aggregation Builder
+ */
+type AggregationUseCaseAddedEvent = ConnectionScoped<{
+  name: 'Aggregation Use Case Added';
+  payload: { drag_and_drop?: boolean; stage_name?: string };
+}>;
+
+/**
+ * This event is fired when user adds/remove a stage or changes the stage name in the stage editor view
+ *
+ * @category Aggregation Builder
+ */
+type AggregationEditedEvent = ConnectionScoped<{
+  name: 'Aggregation Edited';
+  payload: {
+    num_stages?: any | number;
+    editor_view_type?: string | 'stage' | 'text' | 'focus';
+    stage_index?: number;
+    stage_action?: string;
+    stage_name?: string;
+  };
+}>;
+
+/**
+ * This event is fired when user runs the aggregation
+ *
+ * @category Aggregation Builder
+ */
+type AggregationExecutedEvent = ConnectionScoped<{
+  name: 'Aggregation Executed';
+  payload: { num_stages: number; editor_view_type: 'stage' | 'text' | 'focus' };
+}>;
+
+/**
+ * This event is fired when a user cancel a running aggregation
+ *
+ * @category Aggregation Builder
+ */
+type AggregationCanceledEvent = ConnectionScoped<{
+  name: 'Aggregation Canceled';
+  payload: {
+    //
+  };
```

Comment:
```suggestion
  payload: Record<string, never>;
```

Reply:
Not that it makes a big difference, but i find the empty payload a lot more immediate to read, is there a benefit to use the  `Record<string, never>`?

Reply:
eslint should yell at you if you use `{}` because `{}` just means any non-nullish value 🙂